src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -376,6 +376,11 @@ def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
         namespace.attach_os_disk = _get_resource_id(
             namespace.attach_os_disk, namespace.resource_group_name, 'disks', 'Microsoft.Compute')
 
+    if getattr(namespace, 'attach_data_disks', None):
```

Comment:
The `attach_data_disks` field might not be in the `namespace` object, arg-parse behavior? I expected the same like you though. It appears we have several other places also using `getattr` to avoid this